authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -0,0 +1,338 @@
+
+func prepareCheckAccessRequestBody(req *authzv1.SubjectAccessReviewSpec, clusterType, resourceId string, retrieveGroupMemberships bool) (*CheckAccessRequest, error) {
+	/* This is how sample SubjectAccessReview request will look like
+		{
+	    	"kind": "SubjectAccessReview",
+	    	"apiVersion": "authorization.k8s.io/v1beta1",
+	    	"metadata": {
+	        	"creationTimestamp": null
+	    	},
+	    	"spec": {
+	        	"resourceAttributes": {
+	            	"namespace": "default",
+	            	"verb": "get",
+					"group": "extensions",
+					"version": "v1beta1",
+					"resource": "deployments",
+					"name": "obo-deploy"
+	        	},
+				"user": "user@contoso.com",
+				"extra": {
+					"oid": [
+	    				"62103f2e-051d-48cc-af47-b1ff3deec630"
+					]
+	        	}
+	    	},
+	    	"status": {
+	        	"allowed": false
+	    	}
+		}
+
+		For check access it will be converted into following request for arc cluster:
+		{
+			"Subject": {
+				"Attributes": {
+					"ObjectId": "62103f2e-051d-48cc-af47-b1ff3deec630",
+					"xms-pasrp-retrievegroupmemberships": true
+				}
+			},
+			"Actions": [
+				{
+					"Id": "Microsoft.Kubernetes/connectedClusters/extensions/deployments/read",
+					"IsDataAction": true
+				}
+			],
+			"Resource": {
+				"Id": "<resourceId>/namespaces/<namespace name>"
+			}
+		}
+	*/
+	checkaccessreq := CheckAccessRequest{}
+	var userOid string
+	if oid, ok := req.Extra["oid"]; ok {
+		val := oid.String()
+		userOid = val[1 : len(val)-1]
+	} else {
+		return nil, errors.New("oid info not sent from authentication module")
+	}
+
+	if isValidUUID(userOid) {
+		checkaccessreq.Subject.Attributes.ObjectId = userOid
+	} else {
+		return nil, errors.New("oid info sent from authentication module is not valid")
+	}
+
+	if !retrieveGroupMemberships {
+		groups := getValidSecurityGroups(req.Groups)
+		checkaccessreq.Subject.Attributes.Groups = groups
+	}
+
+	checkaccessreq.Subject.Attributes.RetrieveGroupMemberships = retrieveGroupMemberships
+	action := make([]AuthorizationActionInfo, 1)
+	action[0] = getDataAction(req, clusterType)
+	checkaccessreq.Actions = action
+	checkaccessreq.Resource.Id = getScope(resourceId, req.ResourceAttributes)
+
+	return &checkaccessreq, nil
+}
+
+func getNameSpaceScope(req *authzv1.SubjectAccessReviewSpec) (bool, string) {
+	var namespace string = ""
+	if req.ResourceAttributes != nil && req.ResourceAttributes.Namespace != "" {
+		namespace = path.Join(namespaces, req.ResourceAttributes.Namespace)
+		return true, namespace
+	}
+	return false, namespace
+}
+
+func ConvertCheckAccessResponse(body []byte) (*authzv1.SubjectAccessReviewStatus, error) {
+	var (
+		response []AuthorizationDecision
+		allowed  bool
+		denied   bool
+		verdict  string
+	)
+	err := json.Unmarshal(body, &response)
+	if err != nil {
+		glog.V(10).Infof("Failed to parse checkacccess response. Error:%s", err.Error())
+		return nil, errors.Wrap(err, "Error in unmarshalling check access response.")
+	}
+
+	if glog.V(10) {
```

Comment:
Should this be done before `json.Unmarshal` ?

Reply:
This was redundant hence removed this.